examples/petsc/src/petscutils.c

```diff
@@ -154,6 +154,7 @@ PetscErrorCode SetupDMByDegree(DM dm, PetscInt p_degree, PetscInt q_extra, Petsc
     PetscCall(DMGetCoordinateDim(dm, &num_comp_coord));
     PetscCall(PetscFECreateLagrange(comm, dim, num_comp_coord, is_simplex, 1, q_degree, &fe_coords));
     PetscCall(DMSetCoordinateDisc(dm, fe_coords, PETSC_TRUE));
+    PetscCall(DMGetCoordinatesLocalSetUp(dm));
```

Comment:
Hmm, why was this needed? I used this for a bit in Ratel to support different logic on different processes previously, but it lead to a massive amount of headache.

Reply:
It's required for the call to `DMSwarmSetPointCoordinatesRandom` for some reason.

Reply:
Ah, eww.

Reply:
Here is what `DMGetCoordinatesLocalSetUp()` does. We would like to call `DMGetCoordinatesLocal()` to extract the local coordinates without being collective. However, if only the global coordinates exist, we try to make them using GlobalToLocal, which is collective. Thus, to assure non-collective access, we need to first call `DMGetCoordinatesLocalSetUp()`.

Reply:
Right, I've used it before. We've had problems with mismatched non-collective and collective logic before since its easy to screw up.

Reply:
> It's required for the call to `DMSwarmSetPointCoordinatesRandom` for some reason.

Can we put it only in the two swarm driver functions right before we set the mesh DMPlex for  the DMSwarm? That way its not changing the behavior of the other BP examples?

Reply:
Sure, though it shouldn't change any behavior -- if I understand correctly, it just pre-caches something that will be created either way. Maybe I'm wrong there.

Reply:
Yeah, it will behave the same. I'm more thinking on the development side. Since we don't really set up any of our function to be non-collective, I'd rather keep getting an error early if we violate that assumption that all our functions are collective.

Reply:
We could also put it right before the call to the `DMSwarmSetPointCoordinatesRandom`

Reply:
Sure, that works for me since that's a collective operation.